src/optimizer/join_order_optimizer.cpp

```diff
@@ -191,29 +194,231 @@ static void UpdateExclusionSet(JoinRelationSet *node, unordered_set<idx_t> &excl
+		    JoinRelationSet::IsSubset(left_join_relations, info->filters[it]->right_set)){
+			auto right_multiplicities = right->multiplicities;
+			auto right_selectivities = right->selectivities;
+			for (idx_t it3 = 0; it3 < info->filters[it]->left_set->count; it3++) {
+				right_relation_id = info->filters[it]->left_set->relations[it3];
+				right_multiplicity = MinValue(right_multiplicities[right_relation_id], right_multiplicity);
+				right_selectivity = MinValue(right_selectivities[right_relation_id], right_selectivity);
+			}
+			assert(info->filters[it]->right_set->count > 0);
+			auto left_multiplicites = left->multiplicities;
+			auto left_selectivities = left->selectivities;
+			for (idx_t it3 = 0; it3 < info->filters[it]->right_set->count; it3++) {
+				left_relation_id = info->filters[it]->right_set->relations[it3];
+				left_multiplicity = MinValue(left_multiplicites[left_relation_id], left_multiplicity);
+				left_selectivity = MinValue(left_selectivities[left_relation_id], left_selectivity);
+
+				if (right_multiplicity * right_selectivity <= min_cardinality_multiplier) {
+					min_left_mult = left_multiplicity;
+					min_left_sel = left_selectivity;
+					min_right_mult = right_multiplicity;
+					min_right_sel = right_selectivity;
+					relation_id_min_left = left_relation_id;
+					relation_id_min_right = right_relation_id;
+				}
+			}
+		}
+		else if (JoinRelationSet::IsSubset(left_join_relations, info->filters[it]->left_set) &&
+		         JoinRelationSet::IsSubset(right_join_relations, info->filters[it]->right_set)) {
+			assert(info->filters[it]->right_set->count > 0);
+			for (idx_t it3 = 0; it3 < info->filters[it]->right_set->count; it3++) {
+				right_relation_id = info->filters[it]->right_set->relations[it3];
+				right_multiplicity = MinValue(right->multiplicities[right_relation_id], right_multiplicity);
+				right_selectivity = MinValue(right->selectivities[right_relation_id], right_selectivity);
+			}
+			assert(info->filters[it]->left_set->count > 0);
+			auto left_multiplicites = left->multiplicities;
+			auto left_selectivities = left->selectivities;
+			for (idx_t it3 = 0; it3 < info->filters[it]->left_set->count; it3++) {
+				left_relation_id = info->filters[it]->left_set->relations[it3];
+				left_multiplicity = MinValue(left_multiplicites[left_relation_id], left_multiplicity);
+				left_selectivity = MinValue(left_selectivities[left_relation_id], left_selectivity);
+
+				if (right_multiplicity * right_selectivity < min_cardinality_multiplier) {
+					min_left_mult = left_multiplicity;
+					min_left_sel = left_selectivity;
+					min_right_mult = right_multiplicity;
+					min_right_sel = right_selectivity;
+					relation_id_min_left = left_relation_id;
+					relation_id_min_right = right_relation_id;
+				}
+			}
+		}
 	}
+
+//	// this technically should never happen as the right and left multiplicities should decrease
+//	// when iterating through the filters.
```

Comment:
If this should technically never happen, would it be better to do a `D_ASSERT` instead of fixing it like this?

By asserting we can identify bugs rather than letting them slip silently